### PR TITLE
(feat) O3-2325 Implement Lab Order form

### PR DESCRIPTION
**Pull request by brandones**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary
Implements the form shown in the [O3 RefApp Zeplin](https://app.zeplin.io/project/60d59321e8100b0324762e05/screen/62c6bf3e8e5a4119570c1bae).

## Screenshots
Screencast from 2023-08-26 15-14-40

## Related Issue
https://issues.openmrs.org/browse/O3-2325

### CODE REVIEW

File: packages/esm-patient-test-results-app/src/config-schema.ts [hunk 1]
```diff
@@ -28,13 +28,25 @@ export const configSchema = {
...
       'Whether or not to display the print button in the Test Results dashboard. When set to `true`, a print button is shown alongside the panel and tree view content switcher. When clicked, a modal pops up showing a datatable with the available test results. Once the user selects an appropriate date range, they can click on the print button in the modal to print the data',
   },
+  orders: {
+    // TODO: What is this and should it default to null (show all orders)?
+    careSettingUuid: {
+      _type: Type.UUID,
+      _description: 'Only orders corresponding to this care setting will be displayed.',
+      _default: '6f0c9a92-6f24-11e3-af88-005056821db0',
+    },
+    labOrderTypeUUID: {
```
Comment (ibacher) [inline packages/esm-patient-test-results-app/src/config-schema.ts:44]: I think we're stuck with this convention:

```suggestion
    labOrderTypeUuid: {
```

**Review by ibacher — APPROVED** [review]

File: packages/esm-patient-test-results-app/src/config-schema.ts (hunk 1 above)
Comment (brandones) [reply packages/esm-patient-test-results-app/src/config-schema.ts:44]: Good catch, thanks. Personally I like that convention. I think Copilot wrote it the other way and I didn't notice.

File: packages/esm-patient-test-results-app/src/filter/filter-set.tsx [hunk 2]
```diff
@@ -32,7 +33,7 @@ function filterTreeNode(inputValue, treeNode) {
 
 const FilterSet: React.FC<FilterSetProps> = ({ hideFilterSetHeader = false }) => {
   const { roots } = useContext(FilterContext);
-  const config = useConfig();
+  const config: ConfigObject = useConfig();
```
[the remark this answers is not in the document]
Comment (brandones) [reply packages/esm-patient-test-results-app/src/filter/filter-set.tsx:36]: Oh yeah that is preferable, thanks

File: packages/esm-patient-test-results-app/src/config-schema.ts [hunk 3]
```diff
@@ -28,13 +28,25 @@ export const configSchema = {
...
   },
-
   showPrintButton: {
     _type: Type.Boolean,
     _default: true,
     _description:
       'Whether or not to display the print button in the Test Results dashboard. When set to `true`, a print button is shown alongside the panel and tree view content switcher. When clicked, a modal pops up showing a datatable with the available test results. Once the user selects an appropriate date range, they can click on the print button in the modal to print the data',
   },
+  orders: {
+    // TODO: What is this and should it default to null (show all orders)?
```
[the remark this answers is not in the document]
Comment (brandones) [reply packages/esm-patient-test-results-app/src/config-schema.ts:38]: Thanks Ian. Filed https://issues.openmrs.org/browse/O3-2359

**Comment by donaldkibet** [comment]
Hi @brandones, could we also incorporate a field to capture the reason for ordering the lab test?

**Comment by makombe** [comment]
Thanks @brandones  for implementing lab order form. How are we handling lab orders eg Viral load, CD4 count etc which may have two kind of results outcome  i.e qualitative and quantitative which are all represented by different concepts.

**Comment by brandones** [comment]
@donaldkibet Sure, I filed https://issues.openmrs.org/browse/O3-2361 for you. Could you please run it by folks in ux-design-advisory? And then please update the ticket according to what gets suggested there. Thanks.

**Comment by brandones** [comment]
@makombe Not sure I understand the problem. The lab order form allows making an order for a test or panel—it makes an order for a single concept. What are you trying to accomplish?

**Pull request merged**